etcdserver/server.go

```diff
@@ -542,7 +552,29 @@ func (s *EtcdServer) run() {
 		case ep = <-etcdprogc:
 		case m := <-s.msgSnapC:
 			merged := s.createMergedSnapshotMessage(m, ep.appliedi, ep.confState)
+			plog.Noticef("log compaction paused when sending snapshot")
```

Comment:
The sending logic is sophisticated enough that a `EtcdServer.sendSnapshot` function would be good now.